src/libraries/System.Private.CoreLib/src/System/Environment.OSX.cs

```diff
@@ -1,12 +1,63 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics;
 using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
 
 namespace System
 {
     public static partial class Environment
     {
         public static long WorkingSet => (long)(Interop.libproc.GetProcessInfoById(ProcessId)?.ptinfo.pti_resident_size ?? 0);
+
+        /// <summary>
+        /// Get the CPU usage, including the process time spent running the application code, the process time spent running the operating system code,
+        /// and the total time spent running both the application and operating system code.
+        /// </summary>
+        [SupportedOSPlatform("maccatalyst")]
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        public static ProcessCpuUsage CpuUsage
```

Comment:
There should be one implementation shared by all Unix-like systems in Environment.Unix.cs based on `SystemNative_GetCpuUtilization` shim. It is what we are doing on other places that we deal with CPU utilization in CoreLib. I do not think the special casing for OSX, etc is worth it.

Reply:
will look at that. For some reason I thought this implementation is not compatible with OSX. I'll give it a try. Thanks!

Reply:
Done. Hopefully, it will work fine with OSX.